src/util/MergingUtil.js

```diff
@@ -0,0 +1,138 @@
+import ColorShape from "../ColorShape";
+
+export default class MergingUtil {
+
+    static mergeShapes(shape1, shape2) {
+        let shape1V = shape1.vertices;
+        let shape2V = shape2.vertices;
+
+        // remove the last element of each array as this will be the duplicate of the first vertex. Makes it easier to merge
+        shape1V.splice(shape1V.length - 1, 1);
+        shape2V.splice(shape2V.length - 1, 1);
+
+        let shape1Map = this.markSharedVertices(shape1V, shape2V);
+        let shape2Map = this.markSharedVertices(shape2V, shape1V);
+     
+        shape1V = this.removeInteriorVertices(shape1V, shape1Map);
+        shape2V = this.removeInteriorVertices(shape2V, shape2Map);
+
+        let mergedVertices = this.mergeVertices(shape1V, shape2V, shape1Map, shape2Map);
+
+        let shape1Neighbors = shape1.neighbors;
+        let shape2Neighbors = shape2.neighbors;
+        let mergedNeighbors = shape1Neighbors.concat(shape2Neighbors.filter((neighbor) => shape1Neighbors.indexOf(neighbor) < 0));
+    
+        mergedNeighbors.splice(mergedNeighbors.indexOf(shape2.seed), 1);
+        mergedNeighbors.splice(mergedNeighbors.indexOf(shape1.seed), 1);
+        let mergedShape = new ColorShape("black", mergedVertices, mergedNeighbors, shape1.seed);
+        
+        return mergedShape;
+    }
+
+    static markSharedVertices(shape1Vertices, shape2Vertices) {
+        let sharedVertexMap = new Map();
+        for(const vertex of shape1Vertices) {
+            if (this.isVertexIn(shape2Vertices, vertex)) {
+                sharedVertexMap.set(vertex, true);
+            } else {
+                sharedVertexMap.set(vertex, false);
+            }
```

Comment:
Can be shortened to `sharedVertexMap.set(vertex, this.isVertexIn(shape2Vertices, vertex))`